paradox-nakadi-consumer-core/src/main/java/de/zalando/paradox/nakadi/consumer/core/http/handlers/EventUtils.java

```diff
@@ -40,18 +45,19 @@ private EventUtils() { }
 
                 final ArrayNode arrayNode = (ArrayNode) eventsNode;
                 arrayNode.elements().forEachRemaining(element -> {
-                    if (!element.isNull()) {
-                        String rawEvent = null;
-                        try {
+                    String rawEvent = null;
+                    try {
+                        if (isEventOfValidType(element, eventType)) {
```

Comment:
If you change `isEventOfValidType` to `checkEventType` as described below, you do not need to change anything in the control flow structure.